### PR TITLE
Introduce a solidus/version.rb file along with Solidus::VERSION

### PR DESCRIPTION
**Pull request by elia**
**Description**

Provide a Solidus::VERSION constant and use it throughout the core frameworks.
I didn't deprecate anything, but it would make sense to me to increase the gap with Spree, at least for the version number.

**Checklist:**
- [x] I have followed [Pull Request guidelines](https://github.com/solidusio/solidus/blob/master/CONTRIBUTING.md#pull-request-guidelines)
- [x] I have added a detailed description into each commit message
- [x] I have updated Guides and README accordingly to this change (if needed)
- [x] I have added tests to cover this change (if needed)
- [x] I have attached screenshots to this PR for visual changes (if needed)

### CODE REVIEW

File: bin/rails-application-template [hunk 1]
```diff
@@ -17,7 +17,8 @@ if $0 == __FILE__
   exec *command
 end
 
-gem 'solidus'
+require_relative "../core/lib/solidus/version"
+gem 'solidus', "~> #{Spree::VERSION}"
```
Comment (kennyadsl) [inline bin/rails-application-template:21]: This should be `Solidus::Version`?

**Review by kennyadsl — COMMENTED** [review]
I'm fine with this but I think we should also [check the `gem-release` configuration](https://github.com/solidusio/solidus/blob/master/.gem_release.yml#L3).

**Comment by waiting-for-dev** [comment]
Hey, @elia, would you like to retake it? :slightly_smiling_face: If not, can we close it?

**Comment by waiting-for-dev** [comment]
I'll close it for now, but we can reopen it at any time if needed 🙂

**Pull request closed without merging**